app/services/AllowanceService.scala

```diff
@@ -38,18 +38,23 @@ trait AllowanceService {
   }
 
   private[services] def allowanceDataConverter(atsData: AtsData): Allowances = {
-    val allowanceData: DataHolder = atsData.allowance_data.get
+    def payload(key: String): Amount =
+      atsData.allowance_data.flatMap(_.payload.flatMap(_.get(key))).getOrElse(Amount.empty)
+
+    def taxpayerName(key: String): String =
+      atsData.taxPayerData.flatMap(_.taxpayer_name.flatMap(_.get(key))).getOrElse("")
 
     Allowances(
       atsData.taxYear,
-      atsData.utr.get,
-      allowanceData.payload.get("personal_tax_free_amount"),
-      allowanceData.payload.get("marriage_allowance_transferred_amount"),
-      allowanceData.payload.get("other_allowances_amount"),
-      allowanceData.payload.get("total_tax_free_amount"),
-      atsData.taxPayerData.get.taxpayer_name.get("title"),
-      atsData.taxPayerData.get.taxpayer_name.get("forename"),
-      atsData.taxPayerData.get.taxpayer_name.get("surname")
+      atsData.utr.getOrElse(""),
```

Comment:
Would `Option` be a better type here? And, do we need to be able to display a NINO similarly?

Reply:
We use the absence of a utr to mean Paye in the viewModel:
 `val isPaye = utr.isEmpty`

This field is not optional for SA, but yes making this an Option would be better in the now combined ViewModel. Thx.